cli/tests/pcluster/config/test_validators.py

```diff
@@ -2285,47 +2281,68 @@ def test_intel_hpc_architecture_validator(mocker, enabled, architecture, expecte
         "enable_intel_hpc_platform",
         enabled,
         intel_hpc_architecture_validator,
-        expected_message,
+        [],
+        expected_errors,
     )
 
 
 @pytest.mark.parametrize(
-    "base_os, architecture, expected_message",
+    "base_os, architecture, expected_warnings, expected_errors",
     [
         # All OSes supported for x86_64
-        ("alinux2", "x86_64", []),
-        ("centos7", "x86_64", []),
-        ("centos8", "x86_64", []),
-        ("ubuntu1804", "x86_64", []),
+        ("alinux2", "x86_64", [], []),
+        ("centos7", "x86_64", [], []),
+        ("centos8", "x86_64", [], []),
+        ("ubuntu1804", "x86_64", [], []),
         # Only a subset of OSes supported for arm64
-        ("alinux2", "arm64", []),
-        ("centos7", "arm64", ["arm64 is only supported for the following operating systems"]),
-        ("centos8", "arm64", []),
-        ("ubuntu1804", "arm64", []),
+        ("alinux2", "arm64", [], []),
+        (
+            "centos7",
+            "arm64",
+            [
+                "Warning: The aarch64 CentOS 7 OS is not validated for the 6th generation aarch64 instances "
+                "(M6g, C6g, etc.). To proceed please provide a custom_ami, "
+                "for more info see: https://wiki.centos.org/Cloud/AWS#aarch64_notes"
+            ],
+            [],
+        ),
+        ("centos8", "arm64", [], []),
+        ("ubuntu1804", "arm64", [], []),
     ],
 )
-def test_architecture_os_validator(mocker, base_os, architecture, expected_message):
+def test_architecture_os_validator(mocker, base_os, architecture, expected_warnings, expected_errors):
     """Verify that the correct set of OSes is supported for each supported architecture."""
     config_dict = {"cluster": {"base_os": base_os, "architecture": architecture}}
     run_architecture_validator_test(
-        mocker, config_dict, "cluster", "architecture", "base_os", base_os, architecture_os_validator, expected_message
+        mocker,
+        config_dict,
+        "cluster",
+        "architecture",
+        "base_os",
+        base_os,
+        architecture_os_validator,
+        expected_warnings,
+        expected_errors,
     )
 
 
 @pytest.mark.parametrize(
-    "disable_hyperthreading, architecture, expected_message",
+    "disable_hyperthreading, architecture, expected_errors, expected_warnings",
     [
-        (True, "x86_64", []),
-        (False, "x86_64", []),
+        (True, "x86_64", [], []),
```

Comment:
Minor: Since these is no test for warning in this test case, we could have not added the extra `expected_warnings` parameter and use the hardcoded `[]` when calling `run_architecture_validator_test`, similar to the way you did it in `test_instances_architecture_compatibility_validator`.

Reply:
fixed